src/lib/Microsoft.Health.Fhir.Ingest/Data/MeasurementToEventMessageAsyncCollector.cs

```diff
@@ -39,6 +47,38 @@ public async Task AddAsync(IMeasurement item, CancellationToken cancellationToke
             await _eventHubService.SendAsync(eventList, partitionKey, cancellationToken).ConfigureAwait(false);
         }
 
+        public async Task AddAsync(IEnumerable<IMeasurement> items, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(items, nameof(items));
+
+            using (var hasher = _hashCodeFactory.CreateDeterministicHashCodeGenerator())
+            {
+                var submissionTasks = items
+                .GroupBy(m =>
+                {
+                    var partitionKey = hasher.GenerateHashCode(m.DeviceId.ToLower()) % _partitionCount;
```

Comment:
>hasher.GenerateHashCode(m.DeviceId.ToLower()) [](http://example.com/codeflow?start=39&length=45)

It looks like your hash value can be negative, do we need to take the absolute value if the hash?  Not sure how event hub handles a negative partition.

Reply:
The hash value gets converted to a String on the next line. It's the string version that EventHub receives, and they'll determine what partition to place that value one. So we aren't actually telling them what partition to place it on, we are just giving the a key in order to have them make the determination.
Since they should be able to handle any string, the fact that it will contain a negative sign shouldn't matter.